controllers/ebpf/internal/permissions/permissions.go

```diff
@@ -155,14 +155,9 @@ func (c *Reconciler) reconcileOpenshiftPermissions(
 	} else {
 		scc.AllowedCapabilities = AllowedCapabilities
 	}
-	if (desired.EnablePktDrop != nil && *desired.EnablePktDrop) ||
-		(desired.EnableDNSTracking != nil && *desired.EnableDNSTracking) {
+	if helper.IsPktDropEnabled(desired) || helper.IsDNSTrackingEnabled(desired) {
 		scc.AllowHostDirVolumePlugin = true
 	}
-	if (desired.EnablePktDrop != nil && !*desired.EnablePktDrop) &&
```

Comment:
This was added to handle special case where u start with both features on then u just disable one since we don't compare current vs desired we could endup clearing the AllowHostDirVolumePlugin there was an issue about this by @memodi

Reply:
https://issues.redhat.com/browse/NETOBSERV-1191 this one?

Reply:
@msherif1234 `scc`  is created a few lines above with `AllowHostDirVolumePlugin` not initialized , and since it's a `bool` , it means it's `false` already . So I don't see the point of setting it to false here?

Reply:
can u try the steps mentioned in the bug shared above to be sure, the reason I added the explicit false to make scc desired different than actual anyway I kind of forget the details its been awhile just make sure the steps showing in the above issue are fine

Reply:
Just tried to reproduce that old ticket, and it works fine: I get no error on the daemonset, pods are updated as expected, flows are flowing with dns/drops info